### PR TITLE
Replace add segments

### PR DESCRIPTION
**Pull request by Juan-M-V**
# TITLE

## Description

Replace memory.add_segment in hints to use VMProxy methods

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/dict_manager.rs [hunk 1]
```diff
@@ -323,7 +313,7 @@ mod tests {
                 Some(initial_dict)
             ))
         );
-        assert_eq!(segments.num_segments, 1);
+        assert_eq!(vm_proxy.segments.num_segments, 1);
```
Comment (pefontana) [inline src/hint_processor/builtin_hint_processor/dict_manager.rs:316]: Will this fail when we turn vm_proxy.segments into a private field?

**Review by pefontana — CHANGES_REQUESTED** [review]
Nice one @Juan-M-V 
Let's check the testing of private fields and we will be good to go

File: src/hint_processor/builtin_hint_processor/dict_manager.rs (hunk 1 above)
Comment (Juan-M-V) [reply src/hint_processor/builtin_hint_processor/dict_manager.rs]: resolved: https://github.com/lambdaclass/cairo-rs/pull/457/commits/441652db416c59a732477ba43525048061c6a791

Comment (pefontana) [reply src/hint_processor/builtin_hint_processor/dict_manager.rs]: It is a bad idea to add a VMProxy method that is only used in testing.
Wont be better to assert `assert_eq!(vm.segments.num_segments, 1);`?

Comment (Juan-M-V) [reply src/hint_processor/builtin_hint_processor/dict_manager.rs]: Actually resolved(?): https://github.com/lambdaclass/cairo-rs/pull/457/commits/85b1591b4965b500a41972722a4a7f9af85a34c9

**Review by pefontana — APPROVED** [review]

**Review by Oppen — APPROVED** [review]

**Pull request merged**